### PR TITLE
Final Merge | Set up a "Hello world" project as one app with Webpack

### PR DESCRIPTION
**Pull request by michael-duke**
## Set up a "Hello world" project as one app with Webpack | Final Merge

In this PR I have : 

- Implemented a connection between a Ruby on Rails back-end and React front-end.
- Understood the pros and cons of different approaches of connecting Ruby on Rails back-end with React front-end.
- Get a chance to use `Webpack` with `jsbundling-rails` gem
- Install needed node dependencies as described in the previous lesson's tutorial.
- Created a `babel.config.js` in the root folder.
- Updated `webpack.config.js` to include the `babel loader` and image loading configs
- Created a table for storing your `messages` and create 5 different greetings.
- Created an `API endpoint` that selects a random greeting from your table (you will need a controller with an action and Rails route).
- Created a static view that will be the root of your app.
- Created a `store`, an `action`, and a `reducer` that will connect to the API endpoint to get the `random greeting`.
- Displayed the random greeting in my `Greeting` component.
- Removed `React.StrictMode` to avoid double refresh and API fetch reload
## ⚠️ Before running App
- Make sure to run `bundle install` and `npm i`
- Create a  `.env` file with the following content:
```
POSTGRES_USER=postgres
# If you declared a password when creating the database:
POSTGRES_PASSWORD=YourPassword
POSTGRES_HOST=localhost
POSTGRES_DB=Hello_React_Rails_development
POSTGRES_TEST_DB=Hello_React_Rails_test
```
- Now you can run `db:create` , `db:migrate`, `db:seed`
- Finally run `./bin/dev` and visit `http://localhost:3001/`

### CODE REVIEW

File: app/javascript/react/src/App.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import React from 'react'; 
+import Greeting from './components/Greeting';
+
+function App() {
+  return (
+    <Greeting />
+  );
```
Comment (mikethreels) [inline app/javascript/react/src/App.js]: - [x] according to the requirements the `Greeting` component should be setup as a route in the app component.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Michael, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/javascript/react/src/App.js (hunk 1 above)
Comment (michael-duke) [reply app/javascript/react/src/App.js]: OOps.. 😄 I beleive I missed that, let me quickly add `react-router` and set it up for you @mikethreels

Comment (michael-duke) [reply app/javascript/react/src/App.js]: @mikethreels thank you for your feedback I have made the required changes 🚀

**Review by prolajumokeoni — APPROVED** [review]
Hi @michael-duke 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- No linter errors ✅ 
- Git flow was used correctly 🚀 
- Good design 💯 
- Good Pr title and description 💪🏿 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

- Please always follow proper Gitflow for projects that require Gitflow

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by michael-duke** [comment]
Thank you @prolajumokeoni and @mikethreels for your kind feedbacks, happy coding 😄 :rocket:

**Pull request merged**